src/hooks/useLocale.jsx

```diff
@@ -0,0 +1,26 @@
+import { useState, useEffect } from "react";
+import Spanish from "../lang/es-US.json";
+import English from "../lang/en-US.json";
+import defaultLanguage from "../utils/constant";
+
+// detect the language preference of the user's browser
+const useLocale = () => {
+  const locale = navigator.language;
+  const [messages, setMessages] = useState(defaultLanguage);
```

Comment:
🔧  `defaultLanguage` here should be a reference to the messages file, not the language.